tests/runner/src/main.rs

```diff
@@ -67,16 +68,19 @@ struct Args {
     /// If not set, sqlness will build GreptimeDB on the fly.
     #[clap(long)]
     bins_dir: Option<PathBuf>,
+
+    /// Preserve persistent state in the temporary directory.
+    /// This may affect future test runs.
+    #[clap(long)]
+    preserve_state: bool,
 }
 
 #[tokio::main]
 async fn main() {
     let args = Args::parse();
 
-    #[cfg(windows)]
-    let data_home = std::env::temp_dir();
-    #[cfg(not(windows))]
-    let data_home = std::path::PathBuf::from("/tmp");
+    let temp_dir = TempDir::new("sqlness").unwrap();
```

Comment:
`tempfile::Builder` supports prefix.

https://docs.rs/tempfile/latest/tempfile/struct.Builder.html#method.prefix

Here is the usage:
https://github.com/GreptimeTeam/greptimedb/blob/3e9bda326717392945e1b7b4c41864df3ce45517/src/common/test-util/src/temp_dir.rs#L17-L19